app/js/enhancedDate.js

```diff
@@ -1,5 +1,107 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function iffe() {
+  'use strict';
 
+  //variables for all of the functions below declared here
+
+
+  var myDate;
+  var myDayName;
+  var dayNames = ["Sunday", "Monday","Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+  var myMonth;
+  var monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+  var todaysDate;
+
+
+  return {
+
+
+    setDate: function setDate(enterDate) {
+        if (enterDate instanceof Date){
+          myDate = enterDate;
+        } else if (isNaN(enterDate) === false){
+          myDate = enterDate;
+        } else{
+          myDate = new Date();
+        }
+    },
+
+
+    getDate: function getDate(enterDate) {
+         if (enterDate instanceof Date){
+            myDate = enterDate;
+         }else if (typeof enterDate === "number"){
+            myDate = enterDate;
+         }else{
+           myDate = new Date();
+         }
+         return myDate;
+
+    },
+
+    getDayName: function getDayName(enterDate) {
```

Comment:
This function shouldn't take any arguments. It's only purpose is to return the day name of whatever Date object is being stored internally. You have the right idea about how to return the day name, so the additional logic of using the enterDate parameter is unnecessary.